src/frontends/tensorflow_common/src/op/crop_and_resize.cpp

```diff
@@ -3,137 +3,93 @@
+
+    // try to retrive crop_size as the constant, otherwise, we cannot support it
+    Shape crop_sizes;
+    get_const_input(node, 3, &crop_sizes);
+
+    // concatenate boxes and box_ind inputs because
+    // ROIPooling accepts ROIs in a format [batch_id, x_1, y_1, x_2, y_2]
+    // prepare box_ind for futher concatenation
+    auto const_one = make_shared<Constant>(element::i32, Shape{1}, 1);
+    box_ind = make_shared<Unsqueeze>(box_ind, const_one);
+    box_ind = make_shared<Convert>(box_ind, element::f32);
+    boxes = make_shared<Concat>(OutputVector{box_ind, boxes}, 1);
+
+    // boxes are going in the format [y1, x1, y2, x2]
+    // so we need to adjust them to the format [x_1, y_1, x_2, y_2]
+    // use Convolution for the swapping
+    auto new_shape = make_shared<Constant>(element::i32, Shape{4}, vector<int32_t>{-1, 5, 1, 1});
+    boxes = make_shared<Reshape>(boxes, new_shape, false);
+    // kernel for the swapping should be the following:
+    // 1, 0, 0, 0, 0,
+    // 0, 0, 1, 0, 0,
+    // 0, 1, 0, 0, 0,
+    // 0, 0, 0, 0, 1,
+    // 0, 0, 0, 1, 0
+    auto kernel =
+        make_shared<Constant>(element::f32, Shape{5, 5, 1, 1}, vector<float>{1, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 1, 0,
+                                                                             0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 1, 0});
+    boxes = make_shared<Convolution>(boxes,
+                                     kernel,
+                                     Strides{1, 1},
+                                     CoordinateDiff{0, 0},
+                                     CoordinateDiff{0, 0},
+                                     Strides{1, 1});
+    // reshape boxes with swapped coordinates back to the shape [num_rois, 5]
+    auto new_shape2 = make_shared<Constant>(element::i32, Shape{2}, vector<int32_t>{-1, 5});
+    boxes = make_shared<Reshape>(boxes, new_shape2, false);
+
+    image = make_transpose(image, {0, 3, 1, 2})->output(0);
+    Output<Node> roi_pooling = make_shared<ROIPooling>(image, boxes, crop_sizes, 1.0f, "bilinear");
+    roi_pooling = make_transpose(roi_pooling, {0, 2, 3, 1})->output(0);
+    set_node_name(node.get_name(), roi_pooling.get_node_shared_ptr());
+    return {roi_pooling};
+}
 
-        auto res = make_shared<Concat>(ng_crop_outputs, 0);
-        set_node_name(node.get_name(), res);
-        return res->outputs();
-    }
+OutputVector translate_crop_and_resize_op(const NodeContext& node) {
+    // retieve attributes
+    // some cases like non-zero extrapolation_value are not supported
+    auto method = node.get_attribute<string>("method", "bilinear");
+    auto extrapolation_value = node.get_attribute<float>("extrapolation_value", 0.0f);
+
+    TENSORFLOW_OP_VALIDATION(
+        node,
+        method == "nearest" || method == "bilinear",
+        "[TensorFlow Frontend] Inconsistent model: CropAndResize support only bilinear and nearest sampling methods.");
```

Comment:
Inconsistent model? It is more like a non-implemented issue

Reply:
According to the specification there are only two methods: https://www.tensorflow.org/api_docs/python/tf/raw_ops/CropAndResize